ste/xfer-setProperties.go

```diff
@@ -0,0 +1,224 @@
+		return azblob.AccessTierCool
+	case common.EBlockBlobTier.Archive():
+		return azblob.AccessTierArchive
+	case common.EBlockBlobTier.None():
+		return azblob.AccessTierNone
+	default:
+		panic("invalid tier type")
+	}
+}
+
+func getPageBlobTierString(tier common.PageBlobTier) azblob.AccessTierType {
+	switch tier {
+	case common.EPageBlobTier.P10():
+		return azblob.AccessTierP10
+	case common.EPageBlobTier.P15():
+		return azblob.AccessTierP15
+	case common.EPageBlobTier.P20():
+		return azblob.AccessTierP20
+	case common.EPageBlobTier.P30():
+		return azblob.AccessTierP30
+	case common.EPageBlobTier.P4():
+		return azblob.AccessTierP4
+	case common.EPageBlobTier.P40():
+		return azblob.AccessTierP40
+	case common.EPageBlobTier.P50():
+		return azblob.AccessTierP50
+	case common.EPageBlobTier.P6():
+		return azblob.AccessTierP6
+	case common.EPageBlobTier.None():
+		return azblob.AccessTierNone
+	default:
+		panic("Invalid tier type")
+	}
+}
+func SetProperties(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer) {
+	// If the transfer was cancelled, then reporting transfer as done and increasing the bytes transferred by the size of the source.
+	if jptm.WasCanceled() {
+		jptm.ReportTransferDone()
+		return
+	}
+
+	// schedule the work as a chunk, so it will run on the main goroutine pool, instead of the
+	// smaller "transfer initiation pool", where this code runs.
+	id := common.NewChunkID(jptm.Info().Source, 0, 0)
+	cf := createChunkFunc(true, jptm, id, func() {
+		to := jptm.FromTo()
+		switch to.From() {
+		case common.ELocation.Blob():
+			setPropertiesBlob(jptm, p)
+		case common.ELocation.BlobFS():
+			setPropertiesBlobFS(jptm, p)
+		case common.ELocation.File():
+			setPropertiesFile(jptm, p)
+		default:
+			panic("Shouldn't have happened")
+		}
+	})
+	jptm.ScheduleChunks(cf)
+}
+
+func setPropertiesBlob(jptm IJobPartTransferMgr, p pipeline.Pipeline) {
+	info := jptm.Info()
+	// Get the source blob url of blob to set properties on
+	u, _ := url.Parse(info.Source)
+	srcBlobURL := azblob.NewBlobURL(*u, p)
+
+	// Internal function which checks the transfer status and logs the msg respectively.
+	// Sets the transfer status and Reports Transfer as Done.
+	// Internal function is created to avoid redundancy of the above steps from several places in the api.
+	transferDone := func(status common.TransferStatus, err error) {
+		if status == common.ETransferStatus.Failed() {
+			jptm.LogError(info.Source, "SET-PROPERTIES ERROR ", err)
+		} else {
+			jptm.Log(pipeline.LogInfo, fmt.Sprintf("SET-PROPERTIES SUCCESSFUL: %s", strings.Split(info.Destination, "?")[0]))
+		}
+
+		jptm.SetStatus(status)
+		jptm.ReportTransferDone()
+	}
+
+	blockBlobTier, pageBlobTier := jptm.BlobTiers()
+	srcBlobType := jptm.Info().SrcBlobType
+	_ = srcBlobType
+	SetPropertiesAPIOption := jptm.SetPropertiesAPIOption()
+
+	if SetPropertiesAPIOption.TransferTier() {
+		var err error = nil
+		switch srcBlobType {
+		case azblob.BlobBlockBlob:
```

Comment:
This can be a single line. In addition, there is tier safety checking in the STE. Please take a look at the blob senders for usage.

Reply:
added tier validation to code. [https://github.com/Azure/azure-storage-azcopy/pull/1743/commits/5ccf809af418ee4eb75e29136a7f7e0c2c115f02](url)